Cython/Compiler/ExprNodes.py

```diff
@@ -8409,16 +8436,36 @@ def generate_special_parallel_unpacking_code(self, code, rhs, use_loop):
         if len(sequence_types) == 2:
             code.putln("if (likely(Py%s_CheckExact(sequence))) {" % sequence_types[0])
         for i, item in enumerate(self.unpacked_items):
-            code.putln("%s = Py%s_GET_ITEM(sequence, %d); " % (
-                item.result(), sequence_types[0], i))
+            if sequence_types[0] == "List":
+                code.putln("#if !CYTHON_AVOID_THREAD_UNSAFE_BORROWED_REFS")
```

Comment:
We usually try to put special cases first just because it's easier if we need to add other special cases.